components/playlists/browser/playlists_media_file_controller.h

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_PLAYLISTS_BROWSER_PLAYLISTS_MEDIA_FILE_CONTROLLER_H_
+#define BRAVE_COMPONENTS_PLAYLISTS_BROWSER_PLAYLISTS_MEDIA_FILE_CONTROLLER_H_
+
+#include <list>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/files/file_path.h"
+#include "base/macros.h"
+#include "base/memory/weak_ptr.h"
+#include "base/values.h"
+
+namespace base {
+class FilePath;
+class SequencedTaskRunner;
+}  // namespace base
+
+namespace content {
+class BrowserContext;
+}  // namespace content
+
+namespace network {
+class SharedURLLoaderFactory;
+class SimpleURLLoader;
+}  // namespace network
+
+class GURL;
+
+namespace brave_playlists {
+
+// Handle one Playlist at once.
+class PlaylistsMediaFileController {
```

Comment:
why is this `PlaylistsMediaFileController`? Isn't it a `PlaylistMediaFileDownloader`? "Controller" just doesn't have any obvious meaning outside of a specific context (like MVC).